src/impl.cpp

```diff
@@ -686,4 +688,76 @@ SparseIndices Manifold::Impl::VertexCollisionsZ(VecView<const vec3> vertsIn,
     return collider_.Collisions<false, false>(vertsIn);
 }
 
+#ifdef MANIFOLD_DEBUG
+std::ostream& operator<<(std::ostream& stream, const Manifold::Impl& impl) {
+  stream << std::setprecision(17);  // for double precision
+  stream << "# ======= begin mesh ======" << std::endl;
+  stream << "# tolerance = " << impl.tolerance_ << std::endl;
+  // TODO: Mesh relation, vertex normal and face normal
+
+  for (const vec3& v : impl.vertPos_)
+    stream << "v " << v.x << " " << v.y << " " << v.z << std::endl;
+  for (size_t i = 0; i < impl.halfedge_.size(); i += 3)
+    stream << "f " << impl.halfedge_[i].startVert + 1 << " "
+           << impl.halfedge_[i + 1].startVert + 1 << " "
+           << impl.halfedge_[i + 2].startVert + 1 << std::endl;
+  stream << "# ======== end mesh =======" << std::endl;
+  return stream;
+}
+#endif
+
+#ifdef MANIFOLD_EXPORT
+MeshGL64 ImportMeshGL64(std::istream& stream) {
```

Comment:
This function imports the `.obj` file. Note that this is a quick-and-dirty implementation that doesn't really support the actual specification, but good enough that it can parse our outputs. I do not plan to make it spec compliant, nor make the format stable, so this will probably keep internal to us.